src/include/migraphx/op/fp_conversion.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_FP_CONVERSION_HPP
+#define MIGRAPHX_GUARD_OPERATORS_FP_CONVERSION_HPP
+
+#include <array>
+#include <migraphx/op/binary.hpp>
+#include <migraphx/operation.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/config.hpp>
+#include <cmath>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct fp_conversion
+{
+    shape::type_t targe_type = shape::half_type;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.targe_type, "target_type"));
+    }
+
+    std::string name() const { return "fp_conversion"; }
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
+        return {targe_type, inputs.front().lens(), inputs.front().strides()};
+    }
+
+    argument compute(const shape& output_shape, std::vector<argument> args) const
+    {
+        argument result{output_shape};
+        result.visit([&](auto output) {
+            args.front().visit(
+                [&](auto input) { std::copy(input.begin(), input.end(), output.begin()); });
```

Comment:
This wont work for non-standard shapes because of issue #39. The best option would be to use the [`unary`](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/241/files#diff-cf8304b78d60ffd3339e4553e74953aeR11) base class from #241. Then this could be defined as:

```cpp
struct fp_conversion : unary<fp_conversion>
{
    shape::type_t targe_type = shape::half_type;

    template <class Self, class F>
    static auto reflect(Self& self, F f)
    {
        return pack(f(self.targe_type, "target_type"));
    }

    shape compute_shape(std::vector<shape> inputs) const
    {
        check_shapes{inputs, *this}.has(1);
        return {targe_type, inputs.front().lens(), inputs.front().strides()};
    }

    auto apply() const
    {
        return [](auto x) { return x; };
    }
};
```

But #241 hasn't been merged yet.